spark8t/cli/service_account_registry.py

```diff
@@ -42,6 +47,18 @@ class Actions(str, Enum):
     LIST = "list"
 
 
+def create_namespace_if_missing(kube_interface: AbstractKubeInterface, namespace: str):
+    """Create namespace if does not exist."""
+    if not kube_interface.exists(KubernetesResourceType.NAMESPACE, namespace):
+        try:
+            kube_interface.create(KubernetesResourceType.NAMESPACE, namespace)
+        except Exception:
+            logger.warn(
+                f"Namespace {namespace} does not exist, and it could not be created."
+            )
+            raise NamespaceNotFound(namespace)
```

Comment:
Also I believe we haven't caught `NamespaceNotFound` yet in `main`. Therefore, the `raise` here will possibly pass through the `except Exception` block in `main` from where it will be `raise`d again.

Reply:
On this regard, it'd be awesome if we could add an integration test for this scenario (where the SA that is creating the SA doesn't have enough permissions). However I'm not sure if there's a way to do that without involving pods and all complexities.